google/gax/api_callable.py

```diff
@@ -41,59 +41,63 @@
 _MILLIS_PER_SECOND = 1000
 
 
-def _add_timeout_arg(a_func, timeout):
-    """Updates a_func so that it gets called with the timeout as its final arg.
+class _RetryableWrapper(object):
+    """A callable to call with either retrying or timeout parameter."""
 
-    This converts a callable, a_func, into another callable with an additional
-    positional arg.
+    def __init__(self, a_func, settings):
+        """Constructor.
 
-    Args:
-      a_func (callable): a callable to be updated
-      timeout (int): to be added to the original callable as it final positional
-        arg.
+        Args:
+          a_func (callable[[req] resp]): an API call.
+          settings (CallSettings): the base settings for the API call.
+        """
+        self._func = a_func
+        self._settings = settings
 
+    def __call__(self, req, options=None, *args, **kwargs):
+        """The API call, customized by the additional CallOptions.
 
-    Returns:
-      callable: the original callable updated to the timeout arg
-    """
+        Args:
+          req (object): the request object.
+          options (gax.CallOptions): the optional value to customize the
+            calling behavior.
+        """
+        settings = self._settings.merge(options)
```

Comment:
I agree that settings.merge() will be in a single place, and restructured a bit. Please take another look. I am not sure this is the right direction though -- adding more and more inner functions of inner functions.

To be clear, this still determine _page_streamable/_bundleable/else when create_api_call() is invoked, for multiple reasons:
- this structure is actually determined when create_api_call() is called, so this is better and more understandable
- a test case actually require to raise an error for inconsistent bundling/page-streaming for create_api_call(), and I think that's the right thing to keep it.